File: app/src/main/java/org/deiverbum/app/data/entity/relation/LHIntercessionsDM.java
package org.deiverbum.app.data.entity.relation;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.data.entity.LHIntercessionsEntity;
import org.deiverbum.app.data.entity.LHIntercessionsJoinEntity;
import org.deiverbum.app.model.LHIntercession;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
public class LHIntercessionsDM {
    @Embedded
    public LHIntercessionsJoinEntity lhPatristica;

    @Relation(
            parentColumn = "intercessionFK",
            entityColumn = "intercessionID",
            entity = LHIntercessionsEntity.class
    )
    public LHIntercessionsEntity preces;

    public LHIntercession getDomainModel() {
        LHIntercession theModel=new LHIntercession();
        theModel.setIntro(preces.intro);
        theModel.setIntercession(preces.preces);
        return theModel;
    }
}

File: app/src/main/java/org/deiverbum/app/data/entity/relation/LHIntercessionsDM.kt
package org.deiverbum.app.data.entity.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.data.entity.LHIntercessionsEntity
import org.deiverbum.app.data.entity.LHIntercessionsJoinEntity
import org.deiverbum.app.model.LHIntercession

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
class LHIntercessionsDM {
    @JvmField
    @Embedded
    var lhPatristica: LHIntercessionsJoinEntity? = null

    @JvmField
    @Relation(
        parentColumn = "intercessionFK",
        entityColumn = "intercessionID",
        entity = LHIntercessionsEntity::class
    )
    var preces: LHIntercessionsEntity? = null
    val domainModel: LHIntercession
        get() {
            val theModel = LHIntercession()
            theModel.intro = preces!!.intro
            theModel.intercession = preces!!.preces
            return theModel
        }
}